src/test/java/io/gravitee/repository/OrganizationRepositoryTest.java

```diff
@@ -88,10 +90,11 @@ public void shouldFindById() throws Exception {
         Optional<Organization> optional = organizationRepository.findById("DEFAULT-ORG-findById");
         Assert.assertTrue("Organization to find not found", optional.isPresent());
     }
-    
+
     @Test
-    public void shouldFindAll() throws Exception {
-        Set<Organization> allOrganizations = organizationRepository.findAll();
-        Assert.assertTrue("No organization found", !allOrganizations.isEmpty());
+    public void shouldCount() throws Exception {
+        final long count = organizationRepository.count();
+        // Should count 3 organizations (DEFAULT-ORG-create, DEFAULT-ORG-update and DEFAULT-ORG-findById)
+        Assert.assertEquals("Organization count should be 3",3L, count);
```

Comment:
I think it could lead to error if another scenario create other organizations.
maybe modify the assertion to "greater than 3" ? wdyt ?

Reply:
I would prefer not to do such thing, because if I want to test that the count() is working I would expect a fixed value.

Reply:
Yeah when I did that I was not very convinced that it was the better choice...

I think we all agreed that in a perfect situation, loading / removing the data before each test should be handled by another mechanism which does not rely on our own repository class to avoid coupling and side effects.

So, currently, there is no real good solution to test such `count` feature. Options I have are:
- Either I let it like it is and, effectively, there is a coupling between tests (others tests need to be executed in the right order) but it is still easily maintainable (and I prefer having a fixed count to be sure the count is working as expected).
- Either I find a way to delete all organizations then add some new organizations before I invoke `count` method but it requires more efforts and more code to maintain and I think the coupling is even bigger.

What are you suggestions ?

Reply:
For me, what you have done is the way to go as we initialize before each test and tear down after each test. So no problem possible IMO.

Reply:
see https://github.com/gravitee-io/gravitee-repository-test/blob/d41ed52ea4db3a938d26c9c15b854d6e069da081/src/test/java/io/gravitee/repository/config/AbstractRepositoryTest.java#L159